### DISCUSSION TITLE
How to integrate code testing into a GitHub workflow

### DISCUSSION THREAD

**Discussion by Davy238** [community/community#80296] [New to GitHub] [answered]
I am trying to learn GitHub to improve my document version control etc. All seems easy with text and .html files. But I am struggling with program scripts.
So, using GitHub Desktop, I copy/paste my program scripts into both the 'Main' repository and 'Dev' branch using File Explorer. I then work on the program script in Dev. But there comes time when I need to test the script. So in GitHub I select the option of viewing the script file in File Explorer then in FE I open the file in the Powershell IDE. But I cant test it cos it reports file access problems - I suspect that GitHub has locked the file. But the same happens even if I close GitHub Desktop.
Obviously, there is something I dont understand.
Many thanks in advance for any help you can give.

**Comment by MantasTech** [comment, marked as answer]
Hey Davy,

GitHub doesn't lock files, so the issue might be with your IDE or file permissions. Try using your IDE's Git integration for version control instead of File Explorer. Also, ensure no other program is using your script when you run it. 

Let me know if it helped :octocat:

**Reply by Davy238** [reply]
Thanks MantasTech, I used your phrase to perform a Google search on 'PowerShell IDE Git Integration' and received many useful articles. Just need to work through them now!
